rattlesnake/01-a-class-of-students/soln-python-restful/main.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+
+import re,sys
+
+# The "database"
+data = {}
+# The "session"
+session = {}
+
+# Internal functions of the "server"-side application
+def error_state():
+    return "Something wrong", ["get", "default", None]
+
+# The "server"-side application handlers
+def default_get_handler(args):
+    rep = "What would you like to do?"
+    rep += "\n1 - Quit" + "\n2 - Upload File"
+    links = {"1" : ["post", "execution", None], "2" : ["get", "file_form", None]}
+    return rep, links
+
+def quit_handler(args):
+    for filename in data:
+        with open(filename,"w") as f:
+            for row in data[filename]:
+                f.write(','.join(row))
+                f.write("\n")
+
+    sys.exit("Goodbye m'lord!")
```

Comment:
sys.exit with anything other than status code 0, results into failure. So this one will just print "Goodby..." and exit with status code 1.